terraform/aws/analytical-platform-data-engineering-production/10ds/vpc.tf

```diff
@@ -31,8 +31,33 @@ resource "aws_flow_log" "data_engineering_vpc" {
   vpc_id          = module.vpc.vpc_id
 }
 
+data "aws_iam_policy_document" "cloudwatch_kms_key_policy" {
+
+  statement {
+    sid     = "LogGroupKMSPermissions"
+    effect  = "Allow"
+    actions = ["kms:Encrypt", "kms:Decrypt", "kms:ReEncrypt*", "kms:GenerateDataKey*", "kms:DescribeKey"]
+    principals {
+      type        = "Service"
+      identifiers = ["logs.amazonaws.com"]
+    }
+    resources = ["*"]
+  }
+
+  statement {
```

Comment:
This is insufficient to give the admins or the deployment role to ability to manage this key. 
https://docs.aws.amazon.com/kms/latest/developerguide/key-policy-default.html#key-policy-default-allow-administrators